datumaro/plugins/mpii_format.py

```diff
@@ -0,0 +1,128 @@
+# Copyright (C) 2021 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import json
+import os.path as osp
+
+import numpy as np
+
+from datumaro.components.annotation import Bbox, Points, PointsCategories
+from datumaro.components.extractor import (
+    AnnotationType, DatasetItem, Importer, SourceExtractor,
+)
+from datumaro.components.format_detection import FormatDetectionContext
+from datumaro.components.media import Image
+
+
+class MpiiPath:
+    ANNOTATION_FILE = 'mpii_annotations.json'
+    HEADBOXES_FILE = 'mpii_headboxes.npy'
+    VISIBILITY_FILE = 'jnt_visible.npy'
+    POS_GT_FILE = 'mpii_pos_gt.npy'
+
+MpiiPointsCategories = [
+    (0, ['r_ankle']),
+    (1, ['r_knee']),
+    (2, ['r_hip']),
+    (3, ['l_hip']),
+    (4, ['l_knee']),
+    (5, ['l_ankle']),
+    (6, ['pelvis']),
+    (7, ['thorax']),
+    (8, ['upper_neck']),
+    (9, ['head top']),
+    (10, ['r_wrist']),
+    (11, ['r_elbow']),
+    (12, ['r_shoulder']),
+    (13, ['l_shoulder']),
+    (14, ['l_elbow']),
+    (15, ['l_wrist'])
+]
+
+class MpiiExtractor(SourceExtractor):
+    def __init__(self, path):
+        if not osp.isfile(path):
+            raise FileNotFoundError("Can't read annotation file '%s'" % path)
+
+        super().__init__()
+
+        self._categories = { AnnotationType.points:
+            PointsCategories.from_iterable(MpiiPointsCategories) }
```

Comment:
`PointCategories` must point to `LabelCategories` items. From what I see, this dataset has the only skeleton structure for all labels, so it must be just copied for each label. The correct definition should look like:

```python
human_points = PointCategories.Category(
  labels=[<the list of point labels above>],
  joints=[(#head, #body), (#l_arm, #body), (#r_arm, #body), ...] # the list of point joints
)
categories = {
  AnnotationType.label: LabelCategories.from_iterable(['human']),
  AnnotationType.points: PointCategories.from_iterable([(0, human_points)])
}
```

Reply:
Fixed.